c7n/output.py

```diff
@@ -323,6 +322,34 @@ def get_metadata(self):
         return res
 
 
+class JSONFormatter(logging.Formatter):
+    """
+    Convert Logs to JSON for more usable cloudwatch logs
+    https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/FilterAndPatternSyntax.html
+    """
+
+    def format(self, record):
+        json_log_object = {"type": "log",
```

Comment:
the other concern here is that on error this will effectively also be omitting line info and stack traces, if record.exc_info is present, we should also capture that info, as its pretty key use case for logging.

Reply:
yup makes sense. adding those in now. also adding in record.pathname to pair with lineno